crates/router/src/core/payments/helpers.rs

```diff
@@ -2361,6 +2361,54 @@ impl AttemptType {
     }
 }
 
+#[inline(always)]
+pub fn is_manual_retry_allowed(
```

Comment:
There seems to be a bit of logic overlap between this function and 
https://github.com/juspay/hyperswitch/blob/9969c930a9fc0e983f77e38da45710b87e1203d1/crates/router/src/core/payments/helpers.rs#L2148. Can it be moved to a common function?